### ISSUE TITLE
3.3.1 Error Identification - possibly inappropriate sufficient techniques

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#4029]
https://www.w3.org/WAI/WCAG22/Understanding/error-identification.html currently lists two techniques which, arguably, aren't really all that "sufficient"

* [ARIA2: Identifying a required field with the aria-required property](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA2) - this, on its own, has nothing to do with *error identification*. making sure that a field programmatically exposes that it's required doesn't do anything to identify errors. it may help prevent them, but not identify them. at least this isn't *just* using `aria-required`, but also visible text as part of the label, so it will work for more than just AT users. the similar [PDF5: Indicating required form controls in PDF forms](https://www.w3.org/WAI/WCAG22/Techniques/pdf/PDF5) is different as here the required property will also trigger an error validation for non-AT users. maybe this ARIA2 technique should be replaced with one that uses the actual `required` (rather than `aria-required`) attribute, which at least will trigger browser-native validation?
* [ARIA21: Using aria-invalid to Indicate An Error Field](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA21) - this technique only works for assistive technology users, in isolation. it won't help any user who isn't using something (generally AT like a screen reader) that also conveys the `aria-invalid` state, so relying solely on this won't be sufficient for all users

**Comment by patrickhlauke** [comment]
tl;dr: ARIA techniques, on their own/in isolation, are generally not the best "sufficient" techniques as they'll generally only work for AT users...

**Comment by TestPartners** [comment]
On its own, I regard the `required` attribute as insufficient. It isn't supported properly yet, so it is only a possible future technique. Given how long it has been unsupported properly in Chrome and Firefox, it seems doubtful it will ever get fixed.

**Comment by fstrr** [comment]
See also #1773

**Comment by patrickhlauke** [comment]
> On its own, I regard the `required` attribute as insufficient. It isn't supported properly yet, so it is only a possible future technique. Given how long it has been unsupported properly in Chrome and Firefox, it seems doubtful it will ever get fixed.

just on this point ... how is it not supported properly, specifically?

**Comment by TestPartners** [comment]
The error messages used to only be displayed for five seconds in Chrome. It was like this for about 20 years, but it was fixed sometime in 2024. However, they introduced a new issue.

If the page is scrolled in Chrome, the popup error message remains visible and moves with the form control as long as the error message remains within the viewport. However, if the page is scrolled far enough for the popup error message to be outside the viewport, it disappears automatically and does not reappear when the page is scrolled back again. At high zoom levels, the error message may never be seen if its starting position is outside the viewport.

If the page is scrolled in Firefox, the form controls move but the error messages do not, so the error messages lose their visual relationship with the form controls. This has always been the case.
